### PR TITLE
Fe toasts

### PR DESCRIPTION
**Pull request by LukaszBielsko**

### CODE REVIEW

File: frontend/src/authenticated/AddPost.js [hunk 1]
```diff
@@ -29,15 +31,20 @@ const AddPost = props => {
...
 
-    if (savePost) {
-      history.push("/");
+    if (savePost.ok) {
       dispatch(saveAllPosts());
+      postSuccessToast("Post added successfully!");
+      history.push("/");
+    } else {
+      const errors = await savePost.json();
+      errorToast(errors);
```
Comment (MarcinCholewka) [inline frontend/src/authenticated/AddPost.js:47]: add empty line

File: frontend/src/authenticated/AddPost.js [hunk 2]
```diff
@@ -29,15 +31,20 @@ const AddPost = props => {
...
       "POST",
       "/api/posts",
       JSON.stringify(post)
     );
 
-    if (savePost) {
-      history.push("/");
+    if (savePost.ok) {
       dispatch(saveAllPosts());
+      postSuccessToast("Post added successfully!");
```
Comment (MarcinCholewka) [inline frontend/src/authenticated/AddPost.js:43]: why not forward the redirect to the toast and fire it when the toast ends?

File: frontend/src/authenticated/EditPost.js [hunk 3]
```diff
@@ -58,21 +59,22 @@ const EditPost = props => {
...
       "PATCH",
       "/api/me/posts/" + id,
       JSON.stringify(markdownPost)
     );
 
-    if (post) {
+    if (post.ok) {
       dispatch(saveAllPosts());
       history.push(`/posts/${id}`);
+      postSuccessToast("Post updated successfully");
```
Comment (MarcinCholewka) [inline frontend/src/authenticated/EditPost.js:77]: why not forward the redirect to the toast and fire it when the toast ends?

File: frontend/src/utils/customHooks/useUser.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { useSelector } from "react-redux";
+
+const useUser = () => {
+  const user = useSelector(state => state.currentUser);
+  return user;
```
Comment (MarcinCholewka) [inline frontend/src/utils/customHooks/useUser.js:5]: add empty line before return statement

File: frontend/src/utils/posts/CRUDposts.js [hunk 5]
```diff
@@ -21,7 +21,15 @@ export const request = async (action, url, data) => {
...
-  return response.ok;
+  return response;
+
+  // if (response.ok) {
+  //   return true;
+  // } else {
+  //   return {
+  //     data: response.json()
+  //   };
+  // }
```
Comment (MarcinCholewka) [inline frontend/src/utils/posts/CRUDposts.js:32]: ✂️ remove if not needed

File: frontend/src/utils/toasts/errorToast.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
```
Comment (MarcinCholewka) [inline frontend/src/utils/toasts/errorToast.js:2]: I'm not sure, but you can only import this styling once in the component where you used <ToastContainer />

File: frontend/src/utils/toasts/postSuccessToast.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
```
Comment (MarcinCholewka) [inline frontend/src/utils/toasts/postSuccessToast.js:2]: same here

File: frontend/src/utils/toasts/errorToast.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+  if (data.errors.hasOwnProperty("title")) {
+    error = `Title ${data.errors.title}.`;
+  } else {
+    error = "Something went wrong. Please try again later.";
+  }
+
+  toast.error(error, {
+    position: "top-right",
+    autoClose: 6000,
```
Comment (MarcinCholewka) [inline frontend/src/utils/toasts/errorToast.js:15]: 6s??? isn't it too long???

Comment (LukaszBielsko) [reply frontend/src/utils/toasts/errorToast.js:15]: I've checked and it does feel a bit too long ;)

**Pull request merged**